README.md

```diff
@@ -50,13 +50,187 @@ module "postgresql" {
 ```
 
+```hcl
+
+module "opensearch_domain" {
+  source = "github.com/camunda/camunda-tf-eks-module/modules/opensearch"
+
+  domain_name     = "my-opensearch-domain"
+  subnet_ids      = module.eks_cluster.private_subnet_ids
+  security_group_ids = module.eks_cluster.security_group_ids
+  vpc_id          = module.eks_cluster.vpc_id
+  cidr_blocks      = concat(module.eks_cluster.private_vpc_cidr_blocks, module.eks_cluster.public_vpc_cidr_blocks)
+
+  instance_type   = "t3.small.search"
+  instance_count  = 4
+  ebs_volume_size = 100
+
+  advanced_security_enabled = true
+  advanced_security_internal_user_database_enabled = true
+  advanced_security_master_user_name = "admin"
+  advanced_security_master_user_password = "password"
+
+  depends_on = [module.eks_cluster]
+}
+```
+
 #### GitHub Actions
 
 You can automate the deployment and deletion of the EKS cluster and Aurora database using GitHub Actions. Below are examples of GitHub Actions workflows for deploying and deleting these resources.
 
-For more details, refer to the corresponding [EKS Actions README](https://github.com/camunda/camunda-tf-eks-module/blob/main/.github/actions/eks-manage-cluster/README.md) and [Aurora Actions README](https://github.com/camunda/camunda-tf-eks-module/blob/main/.github/actions/aurora-manage-cluster/README.md), [Cleanup Actions README](https://github.com/camunda/camunda-tf-eks-module/blob/main/.github/actions/eks-cleanup-resources/README.md).
+For more details, refer to the corresponding [EKS Actions README](./.github/actions/eks-manage-cluster/README.md), [Aurora Actions README](./.github/actions/aurora-manage-cluster/README.md) and [OpenSearch Actions README](./.github/actions/opensearch-manage-cluster/README.md), [Cleanup Actions README](./.github/actions/eks-cleanup-resources/README.md).
+
+An example workflow can be found in [here](./.github/workflows/test-gha-eks.yml).
+
+#### Advanced usage with IRSA
+
+This documentation provides a step-by-step guide to creating an EKS cluster, an Aurora RDS instance, and an OpenSearch domain with IRSA (IAM Roles for Service Accounts) roles using Terraform modules.
+The modules create the necessary IAM roles and policies for Aurora and OpenSearch. To simplify the configuration, the modules use the outputs of the EKS cluster module to define the IRSA roles and policies.
+
+### Aurora IRSA Role and Policy
 
-An example workflow can be found in https://github.com/camunda/camunda-tf-eks-module/blob/main/.github/workflows/test-gha-eks.yml.
+The Aurora module uses the following outputs from the EKS cluster module to define the IRSA role and policy:
+
+- `module.eks_cluster.oidc_provider_arn`: The ARN of the OIDC provider for the EKS cluster.
+- `module.eks_cluster.oidc_provider_id`: The ID of the OIDC provider for the EKS cluster.
+- `var.account_id`: Your account id
+- `var.aurora_cluster_name`: The name of the Aurora cluster to access
+- `var.aurora_irsa_username`: The username of the user used to access to the AuroraDB
```

Comment:
what you mean is essentially the user available on the database that should probably equal the user defined on creation.
From just the variable name and description, it may sound misleading, like one needs to use an IAM username.

Reply:
Or maybe have to be linked to this: https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.DBAccounts.html

not sure there's any user in the DB that can do IAM authentication by default.